linetools/lists/linelist.py

```diff
@@ -280,7 +307,100 @@ def set_lines(self, verbose=True, use_cache=True, use_ISM_table=True):
         self._data = tmp_tab
         CACHE['data'][key] = self._data
 
-    def subset_lines(self, subset, sort=False, reset_data=False, verbose=False):
+    def set_extra_columns_to_datatable(self, abundance_type='solar', ion_correction='none'):
+        """Sets new convenient columns to the self._data QTable.
+        These include:
+            - `ion_name` : HI, CIII, CIV, etc
+            - `log(w*f)` : np.log10(wrest * fosc)  # in np.log10(AA)
+            - `abundance` : given by [`none`, `solar`]
+            - `ion_correction` : [`none`]
+            - `rel_strength` : log(w*f) + abundance + ion_correction
+
+        This function is only implemented for the following lists: HI, ISM, EUV, Strong
+
+        """
+        good_linelists = ['HI', 'ISM', 'EUV', 'Strong']
+        if self.list not in good_linelists:
+            warnings.warn('Not implemented: will not set relative strength for LineList: {}.'.format(self.list))
+            return
+
+        # Set ion_name column
+        ion_name = [name.split(' ')[0] for name in self.name]
+        self._data['ion_name'] = ion_name
+
+        # Set QM strength as MaskedColumn (self._data['f'] is MaskedColumn)
+        qm_strength = self._data['f'] * (self._data['wrest'].to('AA').value)
+        qm_strength.name = 'qm_strength'
+        self._data['log(w*f)'] = np.log10(qm_strength)
+        # mask out potential nans
+        cond = np.isnan(self._data['log(w*f)'])
```

Comment:
I noticed at least one element in the ISM LineList with `f=0`, so masking is safer.